Intermittent test failure: test_experiment_by_slug_ready_for_review

Seen https://app.circleci.com/pipelines/github/mozilla/experimenter/10488/workflows/8f33f615-0ac7-40f4-a4ea-01da81b83113/jobs/31062


Since we made channel required for mobile, we have to explicitly set the application in basically all tests.

```py
=================================== FAILURES ===================================
___________ TestNimbusQuery.test_experiment_by_slug_ready_for_review ___________
[gw3] linux -- Python 3.9.6 /usr/local/bin/python3

self = <experimenter.experiments.tests.api.v5.test_queries.TestNimbusQuery testMethod=test_experiment_by_slug_ready_for_review>

    def test_experiment_by_slug_ready_for_review(self):
        user_email = "user@example.com"
        experiment = NimbusExperimentFactory.create_with_lifecycle(
            NimbusExperimentFactory.Lifecycles.CREATED
        )
    
        response = self.query(
            """
            query experimentBySlug($slug: String!) {
                experimentBySlug(slug: $slug) {
                    name
                    slug
                    publicDescription
                    readyForReview {
                        message
                        ready
                    }
                }
            }
            """,
            variables={"slug": experiment.slug},
            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
        )
        self.assertEqual(response.status_code, 200, response.content)
        content = json.loads(response.content)
        experiment_data = content["data"]["experimentBySlug"]
        self.assertEqual(experiment_data["name"], experiment.name)
        self.assertEqual(experiment_data["slug"], experiment.slug)
        self.assertEqual(
            experiment_data["publicDescription"], experiment.public_description
        )
>       self.assertEqual(
            experiment_data["readyForReview"], {"message": {}, "ready": True}
        )
E       AssertionError: {'message': {'channel': ['Channel is required for this[28 chars]alse} != {'message': {}, 'ready': True}
E       + {'message': {}, 'ready': True}
E       - {'message': {'channel': ['Channel is required for this application.']},
E       -  'ready': False}

experimenter/experiments/tests/api/v5/test_queries.py:199: AssertionError

----------- coverage: platform linux, python 3.9.6-final-0 -----------
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-1538)

--- a/app/experimenter/experiments/tests/factories/nimbus.py
+++ b/app/experimenter/experiments/tests/factories/nimbus.py
@@ -184,9 +184,7 @@ class NimbusExperimentFactory(factory.django.DjangoModelFactory):
     firefox_min_version = factory.LazyAttribute(
         lambda o: random.choice(list(NimbusExperiment.Version)).value
     )
-    application = factory.LazyAttribute(
-        lambda o: random.choice(list(NimbusExperiment.Application)).value
-    )
+    application = NimbusExperiment.Application.DESKTOP
     channel = factory.LazyAttribute(
         lambda o: random.choice(list(NimbusExperiment.Channel)).value
     )

--- a/app/experimenter/experiments/tests/test_changelog_utils/test_changelog_utils_nimbus.py
+++ b/app/experimenter/experiments/tests/test_changelog_utils/test_changelog_utils_nimbus.py
@@ -166,6 +166,8 @@ class TestNimbusExperimentChangeLogSerializer(TestCase):
 
 
 class TestGenerateNimbusChangeLog(TestCase):
+    maxDiff = None
+
     def setUp(self):
         self.user = UserFactory.create()
 

